### PR TITLE
Piece together multi-message responses

### PR DESCRIPTION
**Pull request by lindgrenj6**

### CODE REVIEW

File: lib/receptor_controller/client/directive_blocking.rb [hunk 1]
```diff
@@ -39,12 +40,12 @@ def wait_for_response(_msg_id)
       end
     end
 
-    # TODO: Review when future plugins with more "response" messages come
-    def response_success(msg_id, message_type, response)
+    def response_success(msg_id, message_type, serial, response)
```
Comment (slemrmartin) [inline lib/receptor_controller/client/directive_blocking.rb:43]: @lindgrenj6 the same methods (success, error etc.) are in directive_non_blocking.rb, need to be changed in the same way

File: lib/receptor_controller/client/directive_blocking.rb [hunk 2]
```diff
@@ -53,25 +54,41 @@ def response_success(msg_id, message_type, response)
...
         response_waiting.signal
       end
     end
 
+    def join_pages
+      data = pages.keys.sort.each_with_object({}) do |num, acc|
+        # If we have pages of data
+        next(pages[num]) unless pages[num][:body].key?("results")
+
+        acc["count"].nil? ? acc["count"] = pages[num][:body]["count"] : acc["count"] += pages[num][:body]["count"]
```
Comment (slemrmartin) [inline lib/receptor_controller/client/directive_blocking.rb:80]: it shouldn't be expected the body is coming from Ansible Tower, this is common gem (at least it's used for satellite check now). 
Moreover "count" is there only if there is successful response

File: lib/receptor_controller/client/directive_blocking.rb [hunk 3]
```diff
@@ -53,25 +54,41 @@ def response_success(msg_id, message_type, response)
...
         self.response_data = nil
         self.response_exception = ReceptorController::Client::ResponseTimeoutError.new("Timeout (#{log_message_common}) [MSG: #{msg_id}]")
         response_waiting.signal
       end
     end
 
+    def join_pages
+      data = pages.keys.sort.each_with_object({}) do |num, acc|
+        # If we have pages of data
+        next(pages[num]) unless pages[num][:body].key?("results")
```
Comment (slemrmartin) [inline lib/receptor_controller/client/directive_blocking.rb:78]: body can be also string in case of error returned by receptor node

**Review by slemrmartin — CHANGES_REQUESTED** [review]
As I would like to keep blocking directives for simple requests, not sure if it's worth doing this PR. It'll need changes to do it in a generic way.

One argument is, that now we're using blocking requests either for POST or for GET one record (not lists) so more responses of type "response" won't come. 

The second argument is, that there is +1 JSON parse and convert for each request

**Comment by syncrou** [comment]
@lindgrenj6 is this still needed? I see a conflict as well.

**Comment by lindgrenj6** [comment]
@syncrou we'll probably need to do something like this eventually - but it should be done in a generic way for both directive blocking/non-blocking. We also need to brainstorm a bit to see if we can come up with a way to not have to parse the json every time. 

Closing for now since we have other (bigger) problems to work on.

**Pull request closed without merging**